plugin/pkg/admission/priority/admission.go

```diff
@@ -63,7 +66,24 @@ var _ = genericadmissioninitializers.WantsExternalKubeClientSet(&priorityPlugin{
 // NewPlugin creates a new priority admission plugin.
 func newPlugin() *priorityPlugin {
 	return &priorityPlugin{
-		Handler: admission.NewHandler(admission.Create, admission.Update, admission.Delete),
+		Handler:      admission.NewHandler(admission.Create, admission.Update, admission.Delete),
+		featureGates: newFeatureGates(),
+	}
+}
+
+// featureGateStatus contains all the featuregates that are required for this plugin
+type featureGateStatus struct {
+	priorityFeatureGateEnabled      bool
+	resourceQuotaFeatureGateEnabled bool
+	criticalPodAnnotationEnabled    bool
+}
+
+// newFeatureGates returns the list of needed featuregates for this plugin to be used.
+func newFeatureGates() featureGateStatus {
```

Comment:
@deads2k - Are you fine with this initialization?

Reply:
wait until https://github.com/kubernetes/kubernetes/pull/84813 merges, then mimic https://github.com/kubernetes/kubernetes/pull/84813/files#diff-3fcca335c60149004f9b76a575070564R123-R127